### PR TITLE
When renaming an imported symbol, rename only in current file

### LINKED ISSUE #14039: Rename of import should be a local rename
**TypeScript Version:**  nightly (2.2.0-dev.20170213)

**Code**

```ts
import { readFileSync } from "fs";
readFileSync("a.txt");
```

Rename `readFileSync` to `read`.

**Expected behavior:**

```ts
import { readFileSync as read } from "fs";
read("a.txt");
```

**Actual behavior:**

```ts
import { read } from "fs";
read("a.txt");
```

And `@types/node/index.d.ts` is modified too!

### PR DESCRIPTION
**Pull request by andy-ms**
Fixes #14039

### CODE REVIEW

File: tests/cases/fourslash/findAllRefsReExportRightNameWrongSymbol.ts [hunk 1]
```diff
@@ -33,5 +33,7 @@ verify.referenceGroups(cFromARanges, [cFromAGroup, aGroup]);
 
 verify.referenceGroups(d, [dGroup, cFromBGroup, bGroup]);
 
-verify.rangesAreRenameLocations([a, cFromA, cUse]);
-verify.rangesAreRenameLocations([b, cFromB, d]);
+verify.renameLocations(a, [a, cFromA, cUse]);
+verify.renameLocations([cFromA, cUse], [{ range: cFromA, prefixText: "x as " }, cUse]);
+verify.renameLocations([b, cFromB], [b, cFromB, d]);
```
Comment (sheetalkamat) [inline tests/cases/fourslash/findAllRefsReExportRightNameWrongSymbol.ts]: Renaming at cFromB shouldn't rename at `b` and it should be prefix `b as` just like import

File: tests/cases/fourslash/findAllRefsReExports.ts [hunk 2]
```diff
@@ -43,13 +43,20 @@ verify.referenceGroups([bang0, bang1], [eBang]);
...
+        case "default":
+            for (const range of ranges) {
+                goTo.rangeStart(defaultC);
+                verify.renameInfoFailed();
+            }
+            break;
+        case "bar":
+            const [r0, r1, ...tail] = ranges;
+            verify.renameLocations(r0, ranges);
+            verify.renameLocations(r1, [{ range: r1, prefixText: "bar as " }, ...tail]);
```
Comment (sheetalkamat) [inline tests/cases/fourslash/findAllRefsReExports.ts]: verify.renameLocations([r1, …tail], [{ range: r1, prefixText: "bar as " }, ...tail]);

**Comment by jwbay** [comment]
This makes sense if the type comes from node_modules, but when I rename something in my application code, I intend it to be a global rename so that names are kept in sync across files. Will there still be a way to do a global rename for a symbol?

**Comment by andy-ms** [comment]
@sheetalkamat Please re-review

**Comment by sheetalkamat** [comment]
@andy-ms I am confused. wasn't this suppose to not include rename locations from transitive cases if symbol originals in node_modules(externalLibrary?)

**Comment by andy-ms** [comment]
I think that should be how it's working now. We transitively follow *exports* from the starting location but not *imports*. If you start inside `node_modules` you'll still be able to rename it everywhere, but if you start in your own file it will just rename the local alias there.

**Pull request merged**